Bounding Box for Selection

arjxn-py opened the pull request:
https://github.com/user-attachments/assets/887b59da-aaf9-4265-b8cc-6c814ebc7d90

Shall fix #468

trungleduc commented on packages/base/src/3dview/mainview.tsx:
```diff
@@ -555,8 +555,12 @@ export class MainView extends React.Component<IProps, IStates> {
+        if (
+          !intersect.object.visible ||
+          !intersect.object.parent?.visible ||
+          intersect.object.name === 'selectionBoundingBox'
```
Let's make `selectionBoundingBox` a constant variable

trungleduc commented:
holding CTRL and click on the object one more time creates a much bigger bounding box. 


https://github.com/user-attachments/assets/e6150fbb-2924-44aa-8383-73b2973f1042

trungleduc commented:
yeah, tested with the lasted preview link and it works well

trungleduc commented:
Thanks!

arjxn-py commented:
There's one more thing wrong related to the color update when changing color. Looks like I've removed something that wasn't meant to be.


https://github.com/user-attachments/assets/784b8a54-bee5-4c95-b33e-4ad9975c5120

arjxn-py commented:
> There's one more thing wrong related to the color update when changing color. Looks like I've removed something that wasn't meant to be.

Fixed.


https://github.com/user-attachments/assets/f7831e22-b4c9-4833-8662-4677400daa4e

arjxn-py commented:
Bot please update snapshots

martinRenou reviewed (changes requested):
I still see a tiny issue, it seems there is still a place where we are setting the selected mesh color to `SELECTED_MESH_COLOR`. Because when I change the color to red, it's actually going to `SELECTED_MESH_COLOR` first.

Actually I would suggest getting rid of the `SELECTED_MESH_COLOR` as we won't be using it anymore. You could define your own constant `BOUNDING_BOXES_COLOR`.


https://github.com/user-attachments/assets/9421ffa8-2ded-425d-a29d-73f9a7a3c635

arjxn-py commented:
Done. Thanks @martinRenou


https://github.com/user-attachments/assets/97b4e025-789d-47b0-89d0-8c1dc01ad123

martinRenou reviewed (approved):
Thanks!
